File: day-1/main.py
totals = [0]

with open('input', 'r') as inputFile:
    for line in inputFile:
        if line == '\n':
            totals.append(0)
        else:
            totals[-1] += int(line)

totals.sort()

print('Part 1: {}'.format(max(totals)))
print('Part 2: {}'.format(sum(totals[-3:])))

File: day-1/main.ts
const totals: number[] = [0];

const fs = require('fs');
const input = fs.readFileSync('input', 'utf-8').split('\n');

input.forEach((line: string) => {
  if (line === '') {
    totals.push(0);
  } else {
    totals[totals.length - 1] += parseInt(line);
  }
});

totals.sort();

console.log(`Part 1: ${totals[totals.length - 1]}`);
console.log(`Part 2: ${totals.slice(-3).reduce((a, b) => a + b, 0)}`);
